tests/testthat/test-expr_name.R

```diff
@@ -13,3 +13,71 @@ test_that("name to_uppercase", {
     c("VAR1", "VAR2")
   )
 })
+
+test_that("name keep", {
+  expect_identical(
+    pl$DataFrame(list(alice = 1:3))$select(
+      pl$col("alice")$alias("bob")$name$keep(),
+      pl$col("alice")$alias("bob")
+    )$columns,
+    c("alice", "bob")
+  )
+})
+
+test_that("name map", {
+  skip_if_not_installed("withr")
+  withr::with_options(
+    list(polars.no_messages = TRUE),
+    {
+      df = pl$DataFrame(list(alice = 1:3))$select(
+        pl$col("alice")$alias("joe_is_not_root")$name$map(\(x) paste0(x, "_and_bob"))
+      )
+      lf = df$lazy()
+      expect_identical(lf$collect()$columns, "alice_and_bob")
+      expect_error(
+        pl$DataFrame(list(alice = 1:3))$select(
+          pl$col("alice")$name$map(\(x) 42) # wrong return
+        ),
+        "was not a string"
+      )
+
+      # TODO: this works but always prints the error message of log("a"), how
+      # can we silence it?
+      # expect_error(
+      #   pl$DataFrame(list(alice=1:3))$select(
+      #     pl$col("alice")$name$map(\(x) log("a"))
+      #   )
+      # )
```

Comment:
How about using `suppressWarnings`

Reply:
It doesn't work, `log("a")` generates an error, not a warning. I'd say we can skip this for now, there's another test above for wrong function output

Reply:
Hmmm, I don't know what this test case is trying to test.
`log()` is an R function, not a polars one, right?
If it doesn't make sense, why not simply delete the comment?

Reply:
We want to test 2 types of errors: 

* the function called in `$map()` works but doesn't return a string -> Polars error
* the function called in `$map()` fails (like `log("a")` -> both R and Polars error, and we want to ensure that it doesn't lead to a panick or another issue with threads

The test that is commented out is for the second case. Problem is that it returns the two errors and `expect_error()` only hides the Polars one, not the R one. Wrapping `expect_error()` in a second `expect_error()` doesn't work either.

To clarify, if we uncomment this test, it correctly passes. The only issue is that it prints the error message of `log("a")` in the output

Reply:
Thanks for the explanation. Okay, I think we can merge it now for now.